linkerd/expose/traverser.go

```diff
@@ -19,11 +20,11 @@ type Traverser struct {
 
 	Client *kubernetes.Clientset
 
-	Logger
+	Logger logger.Handler
 }
 
 // Visit function traverses each of the resource mentioned in the Traverser struct
-func (traverser *Traverser) Visit(f func(runtime.Object, error) error, continueOnError bool) error {
+func (traverser *Traverser) Visit(f func(runtime.Object, string, error) error, continueOnError bool) error {
 	var errs []error
 	for _, res := range traverser.Resources {
 		md := strings.Split(res, " ")
```

Comment:
Should we add more of the code comment with this in detail? Maybe copy the example of the `res`of `traverser.Resources` is good.

Reply:
@Aisuko I agree, will add more comments here, this callback function is a bit vague at the moment. Will make the changes. Thanks :+1:

Reply:
Are we ready to merge here or do we have outstanding updates?

Reply:
@leecalcote, I forgot to close this PR. I moved this PR to meshkit (https://github.com/layer5io/meshkit/pull/18). I have made all the changes there which were suggested here. That function is more generic and was tested to expose grafana and prometheus deployed by istio (as addons) as well as linkerd.

On another note, for PRs like https://github.com/layer5io/meshery/pull/2081 (merged) and https://github.com/layer5io/meshery/pull/2070 (open), we faced the issue to identify that which adapter/service mesh was used to deploy the grafana/prometheus or other addons, can we leverage this `Expose` function to annotate the services that we create which will help us determine which adapter or service mesh was responsible for the service's deployment?